packages/core/src/browser/quick-open/quick-open-service.ts

```diff
@@ -17,6 +17,7 @@
 import { injectable } from 'inversify';
 import { QuickOpenModel } from './quick-open-model';
 import { MessageType } from '../../common/message-service-protocol';
+import { HideReason } from '../../common/quick-pick-service';
```

Comment:
`QuickOpenService` cannot use own clients, it introduce a cyclic dependency

Reply:
You can move `HideReason` to this file instead and rename to `QuickOpenHideReason`.

Reply:
Sorry I don't understand. Why introduces a cyclic dependency? common/quick-pick-service was not require `quick-open-service.ts `

Reply:
@akosyakov If I move `HideReason`  to this file instead and rename to `QuickOpenHideReason`. It will throw warning `Argument of type 'QuickPickHideReason | undefined' is not assignable to parameter of type 'QuickOpenHideReason | undefined'.` when quickpick invoke quickopen.

Reply:
conceptually, quick-pick is a specialization of quick-open, dependencies should be in one direction

either `hide` method should go to quick-pick, or a hide reason should go to quick-open

Reply:
I think `QuickOpenHideReason` is OK，but `quickopen` is browser module, `packages/core/src/common/quick-pick-service.ts` can't require broswer.

Reply:
We should move `quick-open` to common as well to be able to call it from the backend. For now we could just reference and add FIXME comment above import.

Reply:
https://github.com/theia-ide/theia/issues/5797

Reply:
ok, could you update as mentioned here: https://github.com/theia-ide/theia/pull/5766/files#r306687286 and then we merge

Reply:
Done.